pkg/controller/client/client_controller.go

```diff
@@ -333,14 +329,22 @@ func (r *ReconcileClient) processOidcRegistration(ctx context.Context, client *o
 
 	// Attempt to get the Client registration; if it isn't there, create a new one, otherwise, update
 	_, err = r.GetClientRegistration(ctx, client)
+	var verb string
 	if err != nil {
 		reqLogger.Info("Client not found, create new Client")
+		verb = "create"
 		err = r.createClient(ctx, client)
 	} else {
 		reqLogger.Info("Client found, update Client")
+		verb = "update"
 		err = r.updateClient(ctx, client)
 	}
 
+	if err != nil {
+		reqLogger.Error(err, fmt.Sprintf("Error occured while attempting to %s Client", verb))
+		return
```

Comment:
should we return the error as well?

Reply:
This is a naked return from a function with named returns, which means that `return` is the same as `return err` in this context.

Reply:
Example of how this is used here: [A Tour of Go](https://go.dev/tour/basics/7)

Reply:
ok